### ISSUE TITLE
Updating to 1.4.0 world fails to load[Bug]: 

### ISSUE THREAD

**Issue by Spyder1369** [FTBTeam/FTB-Modpack-Issues#152]
### Modpack

Oceanblock

### Modpack version

1.4.0

### Log Files

https://paste.feed-the-beast.com/view/695ec96c

### Describe the bug

Ticking block entity crash when loading world after update. The log has issues with farmers delight tomatoes.

### Steps to reproduce

Unknown, trying to figure out the block that caused the issue at the moment.

### Expected behaviour

World Should load

**Comment by legitswitch** [comment]
Same issue: https://hastebin.com/yebelimuye.rust

**Comment by Spyder1369** [comment]
Ok after some more testing, I downgraded farmers delight and rolled back to my latest back up and it appears to be working fine no issue

**Comment by Raidobw2** [comment]
vectorwing/FarmersDelight#334

TL;DR: Farmer's Delight had a class reorganisation which seemed a little bit subtile. Normally "old" Tomato crops would be erased from the game with the update, but instead they simply crash with a Ticking Block Entity and NoClassDefFound. Sent the issue to the mod maker and waiting on feedback.

-------------------------------------

Permanent fix for those who want it fixed right now and don't mind the extra work (otherwise, wait for us to give you another solution very soon, even update the pack in one day):
For the FTB App or any other launcher.

FTB App: Click on OceanBlock > Click Settings at the left > Click Open Folder at the bottom.

**\>\>\>Open your `saves` folder and copy paste your world elsewhere to keep a backup just in case.<<<**

Open your `saves` folder then open your world folder. Inside the world folder, open the `serverconfig` folder.
Open `forge-server.toml` and change the text from false to true:
```
#Set this to true to remove any TileEntity that throws an error in its update method instead of closing the server and reporting a crash log. BE WARNED THIS COULD SCREW UP EVERYTHING USE SPARINGLY WE ARE NOT RESPONSIBLE FOR DAMAGES.
	removeErroringTileEntities = false
```

Will become => `removeErroringTileEntities = true`.
Save the file and load the game up again.

Once you load in it should work fine, although every Hydroponic Bed you had with Tomato crops or other "old" crops will be gone. THIS WILL DESTROY SOME BLOCKS.
Once you are done with it and seeing the extent of the damage, close the game and revert the changes in the `forge-server.toml` file, change it back to false.

You can give yourself items to repair for what has been broken and use Hydroponic Beds again. The error will not reoccur, this is due to a transition in the mod.
On Singleplayer open the Escape Menu, Click "Open to LAN" and Click the button to enable cheats. This way you can give yourself items back.

**Comment by Raidobw2** [comment]
Sunekaer technical info is in the issue on Farmer's Delight issue tracker if you want to read the "why this happened", although I don't think you could've foreseen this, definitely seems unintended in his mod transition.

**Comment by Raidobw2** [comment]
@Sunekaer
IF = Industrial Foregoing, FD = Farmer's Delight

Summary: FD had a class reorganization which would normally delete old crops correctly from the game. For X reason, development got harder on FD's side and some stuff didn't make it to the changelog (that's why you didn't see it). I'm assuming they had massive restructure in the code, stuff gets heavy sometimes that's fine, we thank them for their mod. IF has a plugin for Harvestable crops that uses Class 1. When FD restructured classes, Class 1 got changed for Class 2. Therefore, IF's integration no longer works, since it was never adapted to work with Class 2. That's the error part.

FD's Author is still trying to decide himself: Keep the new FD structure or rollback the class changes and postpone to 1.17. I told him I can report to IF since that seems like an easy fix (although other modpack makers gotta be careful when they update IF). I also told him to make an API since that seemed like a great idea.

The immediate solution and fix for the modpack is to rollback FD's version to 0.4.6.

If IF implements Class 2 instead of Class 1, we will be able to update both FD and IF at the same time in a later OceanBlock update. That's all for this issue.
